encoders/nlp/TFIDFTextEncoder/__init__.py

```diff
@@ -0,0 +1,36 @@
+__copyright__ = "Copyright (c) 2021 Jina AI Limited. All rights reserved."
+__license__ = "Apache-2.0"
+
+from jina.executors.encoders import BaseEncoder
+import os
+cur_dir = os.path.dirname(os.path.abspath(__file__))
+
+
+class TFIDFTextEncoder(BaseEncoder):
+    """Encode data from a `np.ndarray` (of strings) of length `BatchSize` into
+    a `csr_matrix` of shape `Batchsize x EmbeddingDimension`. 
+
+    :param path_vectorizer: path containing the fitted tfidf encoder object
+    :param args: not used
+    :param kwargs: not used
+    """
+
+    def __init__(self,
+                 path_vectorizer= "./model/tfidf_vectorizer.pickle",
+                 *args,
+                 **kwargs):
+        super().__init__(*args, **kwargs)
+        self.path_vectorizer = path_vectorizer
+
+    def post_init(self):
+        import pickle
+        self.tfidf_vectorizer = pickle.load(open( os.path.join(cur_dir,self.path_vectorizer), "rb"))
+
+    def encode(self, data, *args, **kwargs) -> 'scipy.sparse.csr_matrix':
```

Comment:
can we use a `custom` batching decorator (independent from the one in the core) that ensures that `data` arrives in `batches` and that the results are merged from a single `csr_matrix`?

Reply:
We could but all decorators from core are for numpy arrays, not sparse arrays (we should construct it then, but core does not depend on scipy making this a bit infeasible in core)

Reply:
thats the point, have it here and custom for csr matrix